packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx

```diff
@@ -410,23 +462,18 @@ export const CommandMenu = () => {
                     }
                   }}
                 >
-                  {!matchingCreateCommand.length &&
-                    !matchingNavigateCommand.length &&
-                    !people.length &&
-                    !companies.length &&
-                    !notes.length &&
-                    !opportunities.length && (
-                      <StyledEmpty>No results found</StyledEmpty>
-                    )}
+                  {isNoResults && !isLoading && (
+                    <StyledEmpty>No results found</StyledEmpty>
+                  )}
```

Comment:
To prevent flashes, here is what I would do.

While the user types a query or we fetch the data for this query, we want to keep displaying the old results, the last results the server returned to us. The issue is that the `useSearchRecords` function returns an empty array of records when it starts refetching. We should be able to access the old results until we receive a successful response.

We should also consider the error case. If the previous results couldn't be fetched because of an error, we have nothing to display while fetching the new results. We can display a blank screen or a loading indicator. I don't have design opinions. Maybe @Bonapara has.

I implemented a similar feature here: https://xstatebyexample.com/search-as-you-type/. **We are not using XState in our project, and I'm not suggesting using it. That's just a comparison.** Furthermore, here we are implementing a UI/UX pattern called a "combobox" and we might envision in the future having reusable toolkits to implement that at several places in the app 😄